code/modules/species/species.dm

```diff
@@ -865,8 +862,10 @@ var/global/const/DEFAULT_SPECIES_HEALTH = 200
 		if(31 to 45)	. = 4
 		else			. = 8
 
-/decl/species/proc/post_organ_rejuvenate(var/obj/item/organ/org, var/mob/living/carbon/human/H)
-	if(org && (org.species ? (org.species.species_flags & SPECIES_FLAG_CRYSTALLINE) : (species_flags & SPECIES_FLAG_CRYSTALLINE)))
+// This should only ever be called via the species set on the organ; calling it across species will cause weirdness.
+/decl/species/proc/apply_species_organ_modifications(var/obj/item/organ/org, var/mob/living/carbon/human/H)
+	SHOULD_CALL_PARENT(TRUE)
+	if(species_flags & SPECIES_FLAG_CRYSTALLINE)
```

Comment:
I'm not sure why you're so fixated on that. Its not because its working now that its not going to break something later, and it'll take ages to track down that this thing was forcing crystal on everything.

On spawn some species got bits that aren't meant to be crystalline. And you're making them all crystaline in here. That's just asking for trouble.

Reply:
I disagree with that. I think the species should be free to do basically whatever to organs. Whether that's at `/species` level with this flag or done at `/adherent` level doesn't really matter to me (though I'd probably have done the latter personally). It's easy to debug: "The species modified the organs, did it do so correctly?" If this is _currently inaccurate as written_ then it's a reasonable objection though.

Reply:
The purpose of renaming the proc was to make it more obvious what it was doing. I don't think it's unclear now.

I really, really don't appreciate being called fixated for fixing bugs with existing code rather than removing it wholesale. It is not an unstable dangerous thing that's going to break other things, it's slapping a flag on some organs. 

Species that have bits that shouldn't be crystalline shouldn't be applying the crystalline flag in this proc and I have yet to see it happen. I've said about nine times now that the entire function of this proc is to apply things that are universal to organs that belong to the species.

The species level flag exists because on Bay there were 3+ species using crystal organs. It could be removed here and the flag put into adherent and mantid modpacks.

Reply:
> I disagree with that. I think the species should be free to do basically whatever to organs. Whether that's at `/species` level with this flag or done at `/adherent` level doesn't really matter to me (though I'd probably have done the latter personally). It's easy to debug: "The species modified the organs, did it do so correctly?" If this is _currently inaccurate as written_ then it's a reasonable objection though.

It should, but it probably shouldn't just change the nature of what it was defined to have at definition, without getting some guarantees first? 

There's no mechanism to make an organ aware that if its flag gets reset, it should go back to being crystal/prosthetics besides this external proc that is called by the parent mob only when the parent modifies the organ. 
Thing is, its perfectly legal for organs to clear their status flag when their damage proc is called or for whatever other reasons. But it cannot know its supposed to reset to being crystal or prosthetic unless it was defined like that? And here we're just willingly not defining the organs with the flag. So we're just removing our only shaky guarantee.

This is what I mean here. This is the kind of things that cause really "fun" bugs from experience..

> The purpose of renaming the proc was to make it more obvious what it was doing. I don't think it's unclear now.
> 
> I really, really don't appreciate being called fixated for fixing bugs with existing code rather than removing it wholesale. It is not an unstable dangerous thing that's going to break other things, it's slapping a flag on some organs.
> 

Well, I really really don't like just getting treated like a dummy, and having a maintainer open a competing PR instead of working things out with me, or just asking if you could just take over it or something? That was a bit rude from you imo.
And yeah I used the word fixated. It wasn't meant to be offensive, and I'm sorry if it was, but I didn't know what else to call it?

> Species that have bits that shouldn't be crystalline shouldn't be applying the crystalline flag in this proc and I have yet to see it happen. I've said about nine times now that the entire function of this proc is to apply things that are universal to organs that belong to the species.
> 
> The species level flag exists because on Bay there were 3+ species using crystal organs. It could be removed here and the flag put into adherent and mantid modpacks.

But, unless I'm wrong, there's not only fully crystalline species with the crystalline flag in this PR? Look at the organs lists. Some of the species with the crystalline flag got at least a random machine or biological part. 

For example mantids got this, and you're setting the species flag to crystalline:
```dm
/obj/item/organ/internal/controller
	name = "system controller"
	desc = "A fist-sized lump of complex circuitry."
	icon = 'mods/species/ascent/icons/ascent.dmi'
	icon_state = "plant"
	parent_organ = BP_CHEST
	organ_tag = BP_SYSTEM_CONTROLLER
	surface_accessible = TRUE
	status = ORGAN_PROSTHETIC
	var/obj/item/card/id/id_card = /obj/item/card/id/ascent
```

Also, the crystal flag comes with different damage handling. It could easily make some parts break when they shouldn't?

Reply:
Any species with the flag expects to apply it to all of their organs including internal. The controller is fine as a crystalline organ, it means they can repair it with the tools on hand. Anything that wants to have non-crystalline organs should be handling it post-flag or not applying the flag at all.

The status flag stuff is a totally separate discussion to this PR. I am not opposed to reworking how flags and limb status are handled. I feel like I'm repeating myself a lot at the moment.